torch/_subclasses/fake_tensor.py

```diff
@@ -1031,6 +1032,21 @@ class _DispatchCacheEntry:
     view_idx: Optional[int]
 
 
+@dataclass_slots
+@dataclass(frozen=True)
+class _DispatchCacheEntry:
+    """
+    Entry type for the FakeTensor dispatch cache. It supports two types of outputs
+    1) tensor
+    2) tuple of tensors
+
+    is_output_tuple flag helps in differentiating the return type
+    """
+
+    output_infos: Tuple[_DispatchCacheEntryOutputInfo]
+    is_output_tuple: bool = False
```

Comment:
Is it not sufficient to check if there's only one vs. many?

Reply:
To differentiate between

`return (out,) `

and 

`return out`

Reply:
For HOPs, we can have a case of tuple of single element.

Reply:
Ohhhh!